vms/platformvm/warp/benchmarks/environment_test.go

```diff
@@ -0,0 +1,414 @@
+	ctx               *snow.Context
+	fx                fx.Fx
+	state             state.State
+	atomicUTXOs       avax.AtomicUTXOManager
+	uptimes           uptime.Manager
+	utxosHandler      utxo.Handler
+	txBuilder         p_tx_builder.Builder
+	backend           *executor.Backend
+	validatorsManager pvalidators.Manager
+}
+
+func (*environment) ResetBlockTimer() {
+	// dummy call, do nothing for now
+}
+
+func newEnvironment(b *testing.B, subnetIDToRegister ids.ID) *environment {
+	res := &environment{
+		isBootstrapped: &utils.Atomic[bool]{},
+		config:         defaultConfig(),
+		clk:            defaultClock(),
+	}
+	res.config.TrackedSubnets.Add(subnetIDToRegister)
+	res.isBootstrapped.Set(true)
+
+	dir := b.TempDir()
+	baseDBManager, err := manager.NewLevelDB(
+		dir,
+		nil,
+		logging.NoLog{},
+		version.Semantic1_0_0,
+		"",
+		prometheus.NewRegistry(),
+	)
+	if err != nil {
+		panic(fmt.Errorf("failed to create mempool: %w", err))
+	}
+
+	res.baseDB = versiondb.New(baseDBManager.Current().Database)
+	res.ctx = defaultCtx(res.baseDB)
+	res.fx = defaultFx(res.clk, res.ctx.Log, res.isBootstrapped.Get())
+
+	rewardsCalc := reward.NewCalculator(res.config.RewardConfig)
+	res.atomicUTXOs = avax.NewAtomicUTXOManager(res.ctx.SharedMemory, txs.Codec)
+
+	res.state = defaultState(res.config, res.ctx, res.baseDB, rewardsCalc)
+	res.uptimes = uptime.NewManager(res.state)
+	res.utxosHandler = utxo.NewHandler(res.ctx, res.clk, res.fx)
+	res.txBuilder = p_tx_builder.New(
+		res.ctx,
+		res.config,
+		res.clk,
+		res.fx,
+		res.state,
+		res.atomicUTXOs,
+		res.utxosHandler,
+	)
+
+	res.backend = &executor.Backend{
+		Config:       res.config,
+		Ctx:          res.ctx,
+		Clk:          res.clk,
+		Bootstrapped: res.isBootstrapped,
+		Fx:           res.fx,
+		FlowChecker:  res.utxosHandler,
+		Uptimes:      res.uptimes,
+		Rewards:      rewardsCalc,
+	}
+
+	registerer := prometheus.NewRegistry()
+	res.sender = &common.SenderTest{}
+
+	metrics := metrics.Noop
+
+	res.mempool, err = mempool.NewMempool("mempool", registerer, res)
+	if err != nil {
+		panic(fmt.Errorf("failed to create mempool: %w", err))
+	}
+
+	res.validatorsManager = pvalidators.NewManager(
+		*res.config,
+		res.state,
+		metrics,
+		res.clk,
+	)
+
+	res.blkManager = blockexecutor.NewManager(
+		res.mempool,
+		metrics,
+		res.state,
+		res.backend,
+		res.validatorsManager,
+	)
+
+	return res
+}
+
+func defaultState(
+	cfg *config.Config,
+	ctx *snow.Context,
+	db database.Database,
+	rewards reward.Calculator,
+) state.State {
+	genesisBytes := buildGenesisTest(ctx)
+	state, err := state.New(
+		db,
+		genesisBytes,
+		prometheus.NewRegistry(),
+		cfg,
+		ctx,
+		metrics.Noop,
+		rewards,
+		&utils.Atomic[bool]{},
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// persist and reload to init a bunch of in-memory stuff
+	state.SetHeight(0)
+	if err := state.Commit(); err != nil {
+		panic(err)
+	}
+	state.SetHeight( /*height*/ 0)
+	if err := state.Commit(); err != nil {
+		panic(err)
+	}
```

Comment:
was this code duplicated unintentionally?

Reply:
yes and in a bunch of places (an old faulty merge prolly). Thanks for spotting, fixed